### DISCUSSION TITLE
3.4 🌈

### DISCUSSION THREAD

**Discussion by maread99** [gerrymanoim/exchange_calendars#107] [General]
# What’s Changed

## Calendar Changes

* Added missing holidays for 2020 and 2021 for XBOM calendar (#97) @rajeshyogeshwar
* [XKRX] Update alternative holiday rule to match the revised law since 2021-08-04 (#78) @elbakramer

## Documentation

* Add tutorials and overhaul README for 3.4 (#106) @maread99

## Enhancements
* Add / Rename `ExchangeCalendar` methods (#105) @maread99
* Implement session nanos (#104) @maread99
* Add `trading_index` function (#100) @maread99
* `side` option to define minute bounds / test suite overhaul (#71) @maread99

## Development Enhancements
* DEV: Add a PR template (#99) @gerrymanoim
* Migrate to new test base, 4 (#96) @maread99
* Migrate to new test base, 2, 3 (#93) @maread99
* Revise / Extend `ExchangeCalendarTestBaseNew` (#92) @maread99
* Migrate to new test base, 1 (#91) @maread99
* Migrate to new test base, 0 (#90) @maread99
* Update make_exchange_calendar_test_csv.py (#81) @maread99

## Bug Fixes
* Fix issue #8 (#88) @maread99

## Maintenance
* Bump actions/checkout from 2.3.4 to 2.3.5 (#102) @dependabot
* Update requirements (#101) @maread99
* Tidy (#80) @maread99

<hr /><em>This discussion was created from the release <a href='https://github.com/gerrymanoim/exchange_calendars/releases/tag/3.4'>3.4 🌈</a>.</em>

**Comment by chicago-joe** [comment]
Holy sh!t man, this is absolutely incredible!! 

Wow, seriously blown away. This does in one python package what our entire tech team has struggled to do for our trading software lol...please keep up the incredible work! I'll be passing this around for sure :)

**Reply by maread99** [reply]
Thanks! Glad you like it! Please do spread the word - the more people looking at and contributing to the calendars the more accurately we can maintain them.
